### PR TITLE
Internal Scarcity 3 - The five minutes nerf

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
This PR makes the emergency tanks smaller, lowering the time for one tank to drain to 5/6 minutes instead of the 14 it currently is.

Need to add docs

## Why It's Good For The Game
Having a small emergency tank last 14 minutes is awful balance. Atmos players now have more incentive to fix the air around the station.

## Changelog
:cl:
balance: Due to budget cuts, we are making all emergency tank smaller, so please seek help from your fellow atmosians workers.
/:cl:

### CODE REVIEW

File: code/__DEFINES/obj_flags.dm [hunk 1]
```diff
@@ -42,17 +42,18 @@
...
+#define NOSLIP_ICE (1<<6)  //prevents from slipping on frozen floors
+#define THICKMATERIAL (1<<7) //prevents syringes, parapens and hypos if the external suit or helmet (if targeting head) has this flag. Example: space suits, biosuit, bombsuits, thick suits that cover your body.
+#define VOICEBOX_TOGGLABLE (1<<8) // The voicebox in this clothing can be toggled.
+#define VOICEBOX_DISABLED (1<<9) // The voicebox is currently turned off.
+#define SCAN_REAGENTS (1<<10) // Allows helmets, masks and glasses to scan reagents.
+#define BLOCKS_SHOVE_KNOCKDOWN (1<<11) // Prevents shovies against a dense object from knocking the wearer down.
+#define SNUG_FIT               (1<<12) //Prevents knock-off from things like hat-throwing.
+#define ANTI_TINFOIL_MANEUVER   (1<<13) //Hats with negative effects when worn (i.e the tinfoil hat).
+#define DANGEROUS_OBJECT    (1<<14)     //Clothes that cause a larger notification when placed on a person.
+#define LARGE_WORN_ICON    (1<<15)     //Clothes that use large icons, for applying the proper overlays like blood
```
Comment (Mothblocks) [inline code/__DEFINES/obj_flags.dm:56]: Remove the alignment indenting from everything you touched.

File: code/modules/clothing/masks/gas_filters.dm [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	desc = "piece of filtering cloth to be used with atmospheric gas masks and emergency gas masks"
+	icon = 'icons/obj/clothing/masks.dmi'
+	icon_state = "gas_atmos_filter"
+	var/filter_status = 100
+	var/filter_strenght_high = 10
+	var/filter_strenght_mid = 8
+	var/filter_strenght_low = 5
+	var/filter_efficiency = 0.5
+
+	var/list/gases_to_check = list(
```
Comment (Mothblocks) [inline code/modules/clothing/masks/gas_filters.dm:12]: Having non-static lists on every single item? Bold.

File: code/modules/clothing/masks/gas_filters.dm [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+		/datum/gas/bz = 0,
+		/datum/gas/tritium = 0,
+		/datum/gas/nitryl = 0,
+		/datum/gas/stimulum = 0,
+		/datum/gas/freon = 0,
+		/datum/gas/hypernoblium = 0,
+		/datum/gas/healium = 0,
+		/datum/gas/proto_nitrate = 0,
+		/datum/gas/zauker = 0,
+		/datum/gas/halon = 0
```
Comment (Mothblocks) [inline code/modules/clothing/masks/gas_filters.dm:41]: Create these dynamically.

File: code/modules/clothing/masks/gas_filters.dm [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+		/datum/gas/healium,
+		/datum/gas/proto_nitrate,
+		/datum/gas/halon,
+		/datum/gas/tritium,
+		/datum/gas/zauker
+	)
+
+/obj/item/gas_filter/proc/reduce_filter_status(datum/gas_mixture/breath)
+
+	for(var/gasID in gases_to_check)
```
Comment (Mothblocks) [inline code/modules/clothing/masks/gas_filters.dm:68]: `gas_id`, here and everywhere else.

File: code/modules/clothing/masks/gas_filters.dm [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+	var/list/low_filtering_gases = list(
+		/datum/gas/healium,
+		/datum/gas/proto_nitrate,
+		/datum/gas/halon,
+		/datum/gas/tritium,
+		/datum/gas/zauker
+	)
+
+/obj/item/gas_filter/proc/reduce_filter_status(datum/gas_mixture/breath)
```
Comment (Mothblocks) [inline code/modules/clothing/masks/gas_filters.dm:66]: Document.

File: code/modules/clothing/masks/gasmask.dm [hunk 6]
```diff
@@ -2,25 +2,65 @@
...
+	if(gas_filter.filter_status <= 0)
+		has_filter = FALSE
+		LAZYREMOVE(gas_filters, gas_filter)
+		qdel(gas_filter)
+	return filtered_breath
+
+/obj/item/clothing/mask/gas/proc/get_filter_durability()
+	var/max_filters_durability = LAZYLEN(gas_filters) * 100
+	var/current_filters_durability
+	for(var/obj/item/gas_filter/gas_filter in gas_filters)
```
Comment (Mothblocks) [inline code/modules/clothing/masks/gasmask.dm:48]:
```suggestion
	for(var/obj/item/gas_filter/gas_filter as anything in gas_filters)
```

Unless you, for whatever reason, expect `null` or non-gas filter related stuff to be in this list.

File: code/modules/clothing/masks/gasmask.dm [hunk 7]
```diff
@@ -2,25 +2,65 @@
...
+		current_filters_durability += gas_filter.filter_status
+	var/durability = (current_filters_durability / max_filters_durability) * 100
+	return durability
 
 /obj/item/clothing/mask/gas/atmos
 	name = "atmospheric gas mask"
-	desc = "Improved gas mask utilized by atmospheric technicians. Still not very good at blocking gas flow, but it's flameproof!"
+	desc =  "Improved gas mask utilized by atmospheric technicians. It's flameproof!"
```
Comment (Mothblocks) [inline code/modules/clothing/masks/gasmask.dm:55]:
```suggestion
	desc = "Improved gas mask utilized by atmospheric technicians. It's flameproof!"
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Making no statement on the change, just the code itself.

File: code/modules/clothing/masks/gas_filters.dm (hunk 2 above)
Comment (Mothblocks) [reply code/modules/clothing/masks/gas_filters.dm:12]: Just so I'm not being cryptic--this is horrible on memory.

**Comment by vpellen** [comment]
I don't hate this as much as the previous one, but I'm still not sure what your logic is regarding "why it's good". Why would dependence be motivating? Does it make their job easier? More enjoyable? More interesting? Does it change what their job is? Does it change how they go about doing their job? Because all it seems to do is make it so that other people want them to do their job more, and I'm having a hard time understanding how that's supposed to be a motivating force.

**Comment by kriskog** [comment]
Gas masks filter out and trap particulates, cleaning the air you breathe. They don't block most real gases, like co2 and n2o.

**Comment by XDTM** [comment]
> Gas masks filter out and trap particulates, cleaning the air you breathe. They don't block most real gases, like co2 and n2o.

In an universe with laser guns, hyperspace travel, and bluespace technology, gas masks have not evolved since the twenty-first century

**Comment by maxymax13** [comment]
> 
> 
> > Gas masks filter out and trap particulates, cleaning the air you breathe. They don't block most real gases, like co2 and n2o.
> 
> In an universe with laser guns, hyperspace travel, and bluespace technology, gas masks have not evolved since the twenty-first century

cameras use film, recorders use casettes, most employees only get issued pdas from the 2000s, there are still old radios lying around, 90% of the station is made out of regular corrodable iron, and producing gloves out of thick rubber still is too expensive to meet employee demands. nanotrasen is cheap af

**Comment by wesoda25** [comment]
> Having a small emergency tank last 14 minutes is awful balance.

Could you elaborate more on this?

**Comment by Ghilker** [comment]
> > Having a small emergency tank last 14 minutes is awful balance.
> 
> Could you elaborate more on this?
> 
> Also it would be based if you made gas masks limit visibility. This PR gives people more incentive to powergame them, there should be some drawbacks.

emergency tanks should be used for "emergencies", having a tank last 14 minutes is an awful lot, getting spaced, running in space from sec, shouldn't be something mostly trivial, 5/6 minutes is a long time to recover and re-enter into a save enviroment but short enough so that you can't mindlessy go into hazards.

The limiting masked visibility is something i thought about, i'll see how to do that

**Comment by wesoda25** [comment]
I guess that makes sense, although I do think it’s a little cruel when combined with linda atmos. I guess we’ll have to see if atmos techs do their jobs or not. 

Also I’m pretty sure the degraded helmet limits visibility a bit, you could look at that.

**Comment by Ghilker** [comment]
> A worry I have is that atmos is simply an unfun mechanic to interact with. Suffocation gets pretty fucking tired when gas moves at the speed of light. Combined with linda atmos this change feels really unfair to the players. I’m pretty sure that lemon has commented that he’s currently feeling out the new atmos and is going to decide how to proceed afterwards - I really think you should wait with this PR until after linda atmos has been tweaked (assuming it will be).
> 
> Also I’m pretty sure the degraded helmet limits visibility a bit, you could look at that.

Aye, managed that, now when wearing a gasmask the screen darkens (except for mime and clown masks)

**Comment by Livrah** [comment]
Plasmaman main.

**Comment by ATH1909** [comment]
> > A worry I have is that atmos is simply an unfun mechanic to interact with. Suffocation gets pretty fucking tired when gas moves at the speed of light. Combined with linda atmos this change feels really unfair to the players. I’m pretty sure that lemon has commented that he’s currently feeling out the new atmos and is going to decide how to proceed afterwards - I really think you should wait with this PR until after linda atmos has been tweaked (assuming it will be).
> > Also I’m pretty sure the degraded helmet limits visibility a bit, you could look at that.
> 
> Aye, managed that, now when wearing a gasmask the screen darkens (except for mime and clown masks)

Er, I don't think that tint actually makes you screen darker. IIRC, it's used for determining the vision impediment from stacking sunglasses, welding gear, etc.

Basically, you've made it so that wearing sunglasses under your gas mask will make it about as hard to see as if you were wearing a welding helmet with a flipped down visor.

File: code/modules/mob/living/carbon/carbon.dm [hunk 8]
```diff
@@ -649,7 +649,10 @@
 		become_blind(EYES_COVERED)
 	else if(tinttotal >= TINT_DARKENED)
 		cure_blind(EYES_COVERED)
-		overlay_fullscreen("tint", /atom/movable/screen/fullscreen/impaired, 2)
+		overlay_fullscreen("tint", /atom/movable/screen/fullscreen/impaired, 2, clamp(tinttotal * 100, 0, 255))
+	else if(tinttotal >= TINT_LIGHTER)
+		cure_blind(EYES_COVERED)
+		overlay_fullscreen("tint", /atom/movable/screen/fullscreen/impaired, 1, clamp(tinttotal * 100, 0, 255))
```
Comment (ATH1909) [inline code/modules/mob/living/carbon/carbon.dm:655]: congratulations, you've just made sunglasses semi-blind people

Comment (Ghilker) [reply code/modules/mob/living/carbon/carbon.dm:655]: Sunglasses are tint=2, gas masks are 1.2 to 0, don't wear the two of them at the same time ¯\_(ツ)_/¯

**Comment by ATH1909** [comment]
> I don't hate this as much as the previous one, but I'm still not sure what your logic is regarding "why it's good". Why would dependence be motivating? Does it make their job easier? More enjoyable? More interesting? Does it change what their job is? Does it change how they go about doing their job? Because all it seems to do is make it so that other people want them to do their job more, and I'm having a hard time understanding how that's supposed to be a motivating force.
> 
> Like, when a hole gets blown in the hull, are there really a bunch of atmos techs sitting in the engineering lobby eating honk pockets going "Man, do I really _have_ to fix that breach? People have internals, right? It's probably fine."

This is also something to consider, although to play devil's advocate, atmos techs being more critical to people not dying could make atmos tech players feel more appreciated when continuing to play atmos tech/do their job, which in turn could cause them to stick around longer.

Alternatively, it could just cause more "well, our (possibly nonexistent) engineers didn't bother to set up the SM this shift, screwing the shift over, so I guess it's time to speedcall the shuttle again" moments.

However, I feel like this'd really just cause more raiding of oxygen lockers, no?

**Comment by ATH1909** [comment]
It'd be nice if you could atomize the gas mask stuff out of this PR.

File: code/modules/mob/living/carbon/carbon.dm (hunk 8 above)
Comment (ATH1909) [reply code/modules/mob/living/carbon/carbon.dm:655]: isn't TINT_LIGHTER 1

Comment (ATH1909) [reply code/modules/mob/living/carbon/carbon.dm:655]: also sunglasses have tint=1, not 2

Comment (Fikou) [reply code/modules/mob/living/carbon/carbon.dm:655]: this will affect using sunglasses by themselves

Comment (Ghilker) [reply code/modules/mob/living/carbon/carbon.dm:655]: right sunglasses have tint=1

and yes you'll get affected for wearing sunglasses inside

File: code/modules/clothing/masks/gasmask.dm [hunk 9]
```diff
@@ -64,6 +111,7 @@
 	icon_state = "plaguedoctor"
 	inhand_icon_state = "gas_mask"
 	armor = list(MELEE = 0, BULLET = 0, LASER = 2,ENERGY = 2, BOMB = 0, BIO = 75, RAD = 0, FIRE = 0, ACID = 0)
+	tint = 0
```
Comment (RaveRadbury) [inline code/modules/clothing/masks/gasmask.dm:114]: I don't get why you're going to keep showing this preference to the plague doctor mask, but not to any of the other costume gasmask children. Like, plaguedoctor masks are super obfuscating visually, but you've decided that the owl mask, tiki mask, carp mask, bounty hunter mask, and cyborg visor ***should*** have a tint?

Thanks for minding the performer job masks.

Comment (RaveRadbury) [reply code/modules/clothing/masks/gasmask.dm:114]: > Aye, managed that, now when wearing a gasmask the screen darkens (except for mime and clown masks)

This makes your choice to save plague doctor masks even more confusing. Is this like a mask that you enjoy wearing and plan to be making use of? Please show the other costume masks the same level of care that you are giving this one.

**Comment by RaveRadbury** [comment]
> It'd be nice if you could atomize the gas mask stuff out of this PR.

Yeah when I saw the comment on the last version of this I assumed you were just nerfing emergency internals amount, which sounds like an interesting idea that I would be in favor of trying.

**Comment by Zytolg** [comment]
I honestly don't see what this accomplishes besides encouraging players to just go and replace their oxytanks out of the lockers at the earliest possible convenience, and therefore creating scarcity on the o2 front when things go south. It already takes several minutes to repair breaches, and in my own experience, I find myself grabbing tanks to swap out my now empty O2 tank while waiting for the air to refill.

From a mapppers standpoint, I wouldn't be surprised if after this went through, we had issues requesting us to add more spare tanks to the oxylockers because there aren't enough and people keep dying of oxyloss.

File: code/modules/clothing/masks/gasmask.dm (hunk 9 above)
Comment (Ghilker) [reply code/modules/clothing/masks/gasmask.dm:114]: oh it's a tint i missed when i was doing code, first i had it like this for all masks, but then i changed things around and forgot about this, i'll remove the tint=0 from plague masks

**Comment by Ghilker** [comment]
> > It'd be nice if you could atomize the gas mask stuff out of this PR.
> 
> Yeah when I saw the comment on the last version of this I assumed you were just nerfing emergency internals amount, which sounds like an interesting idea that I would be in favor of trying.

I mean, i could atomize the gas filtering thing from this so this PR would only be about the emergency tank nerf, if requested

**Comment by Pugie** [comment]
roleplayer pr

**Comment by RaveRadbury** [comment]
> I mean, i could atomize the gas filtering thing from this so this PR would only be about the emergency tank nerf, if requested

Nerfing the air tank time only has a better chance of going through, imo.

**Comment by optimumtact** [comment]
It's unrelated but I have wanted gasmasks to obscure vision for some time to stop people wearing them in all circumstances

**Comment by SkeletalElite** [comment]
It's far too easy to fuck up the atmosphere and far too much of a PITA to fix it for this to be good IMO.

**Comment by kriskog** [comment]
> 
> 
> It's unrelated but I have wanted gasmasks to obscure vision for some time to stop people wearing them in all circumstances

Making them incompatible with eyewear is probably a good compromise, and would make a lot of sense. Make sure the security one cannot be raised while wearing huds, or declassify it as a gas mask entirely.

**Comment by Ghilker** [comment]
Ok i'm going to atomize this PR to make it easier to review, so this one will only deal with the reduced emergency tanks, i'll make another PR with the gas masks changes

**Comment by maxymax13** [comment]
> 
> 
> It's unrelated but I have wanted gasmasks to obscure vision for some time to stop people wearing them in all circumstances

i can't believe oranges is trying to make peak tider fashion inconvenient :(

**Comment by RaveRadbury** [comment]
> i can't believe oranges is trying to make peak tider fashion inconvenient :(

There's usually a price to pay for style and looking good.

**Comment by cacogen** [comment]
People don't wear these very often nowadays (unless it's part of their character). If you want to obscure your face, there are less obvious ways of doing it and if you want to wear a breath mask at all times, then the type in your box can be toggled off to eat instead of having to remove it.

**Comment by RaveRadbury** [comment]
@Ghilker 
image
This is left over, gotta clear it

**Comment by wesoda25** [comment]
> People don't wear these very often nowadays (unless it's part of their character). If you want to obscure your face, there are less obvious ways of doing it and if you want to wear a breath mask at all times, then the type in your box can be toggled off to eat instead of having to remove it.

They go like hotcakes on the lrp servers

**Comment by LemonInTheDark** [comment]
I'm not totally sure we should be doing this at this stage, since the station can already get quite cold quite quickly, so even if you do have internals you're not really safe. I suppose I'm making an assumption here, is your target to effect how people react to o2 loss? or toxins in the air supply

**Comment by Ghilker** [comment]
This is mainly targeting deficit of o2 and toxins in the air, being able to survive in a toxic ambient for 14 minutes with a small tank doesn't make it balanced at all, emergency tanks should be just that, emergency
Coldness won't be affected since there isn't much difference with the tanks from before and after this PR

**Comment by Ghilker** [comment]
> @Ghilker
> image
> This is left over, gotta clear it

is not showing up for me, it's already removed

**Comment by RaveRadbury** [comment]
I'm still seeing it on my end.

**Comment by Twaticus** [comment]
same

**Comment by cacogen** [comment]
I like scarcity too. It makes everything more valuable and meaningful. But not in this. The station is suffocation central at the moment with atmos as fast as it is (and people either not playing it, not having access, or not knowing how to do it) and these tanks don't last very long because of it.

**Comment by YakumoChen** [comment]
I would think this is a good idea if, say, oxygen tanks were easily and readily refilled by anyone outside of engi/sci, but in actuality the vast majority of the station is going to run out of air and be forced to scavenge for tanks in whatever lockers are lying around because there's no free refills on air. 

Any breach is going to be that much quicker of a shuttle call because people will start dying of oxygen loss after about 10 minutes and you can't just epi yourself or whatever anymore for that free 'I'm about to be fucked but gimmie another minute' pick me up when they realize their tank's gone dry.

People don't appreciate Atmosia anyway so this won't exactly change public opinion on that when they're choking for air and they have a monopoly on o2 they can't easily distribute.

**Comment by Jhyrachy** [comment]
> I would think this is a good idea if, say, oxygen tanks were easily and readily refilled by anyone outside of engi/sci, 


This could be a fun idea for a change:
remove air closet and replace them for a new machine, the Air Dispenser.  

It must be connected to air pipline
At round start it could have the same items of a closet, but also in addition using a tank on it will be refilled (like a fire extinguisher on a water tank)

Extra stuff: could possibly be used to help repressurizing and refilling rooms

Pro: new stuff to build, less clutter, more easy access too air, more eco friendly since you can refill your equipment

**Comment by RaveRadbury** [comment]
I played one of my first rounds on Tramstation and I latejoined to plasma in the air, as a cook I had no idea where atmos was or if they were doing their job, plus I had to get to the kitchen. After some time I was able to remove my O2 mask but I still had to deal with low levels of N2O for the remainder of the shift I think.

This would have been far more unpleasant with a 5-minute air tank, I do not believe that the atmos will solve an air contamination issue within five minutes, *if there is an atmos tech on station in the first place*. A 66% reduction in emergency internal air is too big of a first step, This should be a nerf to 10-minutes instead of 5.

**Comment by Learner10** [comment]
> 
> I do not believe that the atmos will solve an air contamination issue within five minutes, _if there is an atmos tech on station in the first place_. A 66% reduction in emergency internal air is too big of a first step, This should be a nerf to 10-minutes instead of 5.

Gotta agree. I tend to encounter situations where air tanks end up becoming scarce quite frequently because someone left a window open and froze the station. Fact of the matter is, the emergency air tanks aren't just used for low pressures. They're used for temperature differences as well, _so your lungs do not freeze_. And considering how 30-40% the rounds is just nonstop fire alarms, this means that on a crew of 40, at least 10-20 of them are using an emergency air tank to avoid the slowdown from a large tank slotted on their back, or are losing one of their two hands just to carry a tank. In fact...

This level of a nerf is going to make conscription kits a daily purchase so people can get the "large air tank holder that doesn't have slowdown" clothing, and a free gas mask.

Other things to note, if you're going to do this, understand that this also makes the emergency cargo crate purchases way weaker, since you're taking 27 minutes of oxygen out from each crate.

And really... why not just make each emergency oxygen can just START with less air inside, but be able to be filled up to their current maximum value? That's a nerf that doesn't say 'please die' to the players who didn't choose to be an engineer for their extended air tanks.

**Comment by RaveRadbury** [comment]
I think it might be worthwhile to compare atmos to the kitchen. The kitchen is easily accessible, has donk pockets, fridges that can be opened via violence, silicon, or HOP, and we have countless vendors all over the station. Cargo can order food, too. We have intense redundancy and failsafe for our food system, despite hunger not being fatal.

How does atmos compare? Is it occupied as frequently as the kitchen is? How about atmos' accessibility, especially in terms of the security issues that come with handing out access to it. Why is atmos set up in such a way that it does not require player intervention? What are the current consequences of no atmos techs, and what would be the consequences in the aftermath of this change? Do we have oxygen redundancy that is comparable to food redundancy?

I'd really like to believe in the friendly atmosian that sets up a tank filling station at the start of every round, but I'm not sure even the _wiki_ informs new players about such a feature beyond "yeah you can fill air tanks".

**Comment by cacogen** [comment]
It is so hard to get inside to turn up the pump to the external ports or get someone with access to do it.

**Comment by kriskog** [comment]
There is no "tank filling station" as those pumps are hooked to air mix as opposed to pure oxygen. They allow you to fill a pump you can move around to refill areas without air.

**Pull request closed without merging**